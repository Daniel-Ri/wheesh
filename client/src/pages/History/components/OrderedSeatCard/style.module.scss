@import '@styles/base';

.card {
  @include directionX;
  justify-content: space-between;

  .leftCard {
    flex-grow: 1;
    @include directionY;
    gap: 0.25rem;

    .name {
      font-size: 70%;
    }

    .description {
      font-size: 60%;
      color: $color-cool-gray;
      @include directionX;
      gap: 0.5rem;

      .seatClass {
        text-transform: capitalize;
      }
    }
  }

  .price {
    font-size: 70%;
  }

  .iconButton {
    font-size: 100%;
    color: $color-strawberry-red;
  }
}