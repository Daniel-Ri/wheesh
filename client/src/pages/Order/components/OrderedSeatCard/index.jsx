import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import QRCode from 'qrcode';
import { censorIdCard, formatRupiah } from '@utils/handleValue';
import { useEffect, useState } from 'react';
import { Dialog } from '@mui/material';
import classes from './style.module.scss';

const OrderedSeatCard = ({ orderedSeat, addLine }) => {
  const [open, setOpen] = useState(false);
  const [imageUrl, setImageUrl] = useState('');

  const handleClose = () => {
    setOpen(false);
  };

  const handleClickCard = () => {
    setOpen(true);
  };

  useEffect(() => {
    const generateQrCode = async () => {
      try {
        const text = JSON.stringify({ id: orderedSeat.id, secret: orderedSeat.secret });
        const response = await QRCode.toDataURL(text);
        setImageUrl(response);
      } catch (error) {
        // error
      }
    };

    if (open) generateQrCode();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [open]);

  return (
    <>
      <div data-testid="OrderedSeatCard" className={classes.card} onClick={handleClickCard}>
        <div className={classes.leftCard}>
          <div className={classes.name}>{orderedSeat.name}</div>
          <div className={classes.description}>
            <div>
              <FormattedMessage id="app_id_card" />
            </div>
            <div>{censorIdCard(orderedSeat.idCard)}</div>
          </div>
          <div className={classes.description}>
            <div>
              <FormattedMessage id="app_coach" /> {String(orderedSeat.Seat.Carriage.carriageNumber).padStart(2, '0')}
            </div>
            <div>|</div>
            <div className={classes.seatClass}>{orderedSeat.Seat.seatClass} class</div>
            <div>{orderedSeat.Seat.seatNumber}</div>
          </div>
        </div>
        <div className={classes.price}>{formatRupiah(orderedSeat.price)}</div>
      </div>
      {addLine && <hr />}

      <Dialog open={open} onClose={handleClose}>
        <div className={classes.dialog}>
          <div className={classes.content}>
            <h2>
              <FormattedMessage id="app_qr_ticket" />
            </h2>
            <img src={imageUrl} alt="" />
            <div className={classes.name}>{orderedSeat.name}</div>
            <div className={classes.description}>
              <div>
                <FormattedMessage id="app_id_card" />
              </div>
              <div>{censorIdCard(orderedSeat.idCard)}</div>
            </div>
            <div className={classes.description}>
              <div>
                <FormattedMessage id="app_coach" /> {String(orderedSeat.Seat.Carriage.carriageNumber).padStart(2, '0')}
              </div>
              <div>|</div>
              <div className={classes.seatClass}>{orderedSeat.Seat.seatClass} class</div>
              <div>{orderedSeat.Seat.seatNumber}</div>
            </div>
          </div>
        </div>
      </Dialog>
    </>
  );
};

OrderedSeatCard.propTypes = {
  orderedSeat: PropTypes.object.isRequired,
  addLine: PropTypes.bool,
};

export default OrderedSeatCard;
