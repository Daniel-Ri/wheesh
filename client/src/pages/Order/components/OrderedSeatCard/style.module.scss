@import '@styles/base';

.card {
  @include directionX;
  justify-content: space-between;
  cursor: pointer;

  .leftCard {
    flex-grow: 1;
    @include directionY;
    gap: 0.25rem;

    .name {
      font-size: 70%;
    }

    .description {
      font-size: 60%;
      color: $color-cool-gray;
      @include directionX;
      gap: 0.5rem;

      .seatClass {
        text-transform: capitalize;
      }
    }
  }

  .price {
    font-size: 70%;
  }

  .iconButton {
    font-size: 100%;
    color: $color-strawberry-red;
  }
}

.card:hover {
  font-weight: 500;
}

.dialog {
  width: 14rem;
  padding: 1rem;
  @include directionY;

  .content {
    @include directionY;

    h2 {
      text-align: center;
      color: $color-marine-blue;
      font-size: 120%;
      margin-bottom: 0.5rem;
    }

    img {
      align-self: center;
      width: 10rem;
      height: 10rem;
    }

    .name {
      font-size: 75%;
    }

    .description {
      font-size: 65%;
      color: $color-cool-gray;
      @include directionX;
      gap: 0.5rem;

      .seatClass {
        text-transform: capitalize;
      }
    }
  }
}