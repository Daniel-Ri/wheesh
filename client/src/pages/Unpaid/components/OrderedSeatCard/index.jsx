import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { censorIdCard, formatRupiah } from '@utils/handleValue';
import classes from './style.module.scss';

const OrderedSeatCard = ({ orderedSeat, addLine }) => (
  <>
    <div data-testid="OrderedSeatCard" className={classes.card}>
      <div className={classes.leftCard}>
        <div className={classes.name}>{orderedSeat.name}</div>
        <div className={classes.description}>
          <div>
            <FormattedMessage id="app_id_card" />
          </div>
          <div>{censorIdCard(orderedSeat.idCard)}</div>
        </div>
        <div className={classes.description}>
          <div>
            <FormattedMessage id="app_coach" /> {String(orderedSeat.Seat.Carriage.carriageNumber).padStart(2, '0')}
          </div>
          <div>|</div>
          <div className={classes.seatClass}>{orderedSeat.Seat.seatClass} class</div>
          <div>{orderedSeat.Seat.seatNumber}</div>
        </div>
      </div>
      <div className={classes.price}>{formatRupiah(orderedSeat.price)}</div>
    </div>
    {addLine && <hr />}
  </>
);

OrderedSeatCard.propTypes = {
  orderedSeat: PropTypes.object.isRequired,
  addLine: PropTypes.bool,
};

export default OrderedSeatCard;
