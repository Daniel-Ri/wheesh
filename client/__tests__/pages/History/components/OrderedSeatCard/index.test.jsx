import { render } from '@utils/testHelper';
import OrderedSeatCard from '@pages/History/components/OrderedSeatCard';

let wrapper;

const mockProps = {
  orderedSeat: {
    id: 1721,
    orderId: 9,
    seatId: 361,
    price: 200000,
    gender: 'Male',
    dateOfBirth: '1951-08-23T01:28:44.000Z',
    idCard: '1217434274792992',
    name: 'John Ganteng',
    email: '1buyf4bn@example.com',
    Seat: {
      id: 361,
      carriageId: 5,
      seatNumber: '11D',
      seatClass: 'economy',
      Carriage: {
        id: 5,
        trainId: 1,
        carriageNumber: 5,
      },
    },
  },
  addLine: true,
};

beforeEach(() => {
  wrapper = render(<OrderedSeatCard {...mockProps} />);
});

describe('Component OrderedSeatCard of History page', () => {
  test('Renders OrderedSeatCard component', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('OrderedSeatCard')).toBeInTheDocument();
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
