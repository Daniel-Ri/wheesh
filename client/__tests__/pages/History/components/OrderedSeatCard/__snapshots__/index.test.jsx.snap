// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component OrderedSeatCard of History page Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        data-testid="OrderedSeatCard"
      >
        <div
          class="leftCard"
        >
          <div
            class="name"
          >
            John Ganteng
          </div>
          <div
            class="description"
          >
            <div>
              Kartu Identitas
            </div>
            <div>
              1217****92
            </div>
          </div>
          <div
            class="description"
          >
            <div>
              Gerbong
               
              05
            </div>
            <div>
              |
            </div>
            <div
              class="seatClass"
            >
              economy
               class
            </div>
            <div>
              11D
            </div>
          </div>
        </div>
        <div
          class="price"
        >
          Rp 200.000
        </div>
      </div>
      <hr />
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      data-testid="OrderedSeatCard"
    >
      <div
        class="leftCard"
      >
        <div
          class="name"
        >
          John Ganteng
        </div>
        <div
          class="description"
        >
          <div>
            Kartu Identitas
          </div>
          <div>
            1217****92
          </div>
        </div>
        <div
          class="description"
        >
          <div>
            Gerbong
             
            05
          </div>
          <div>
            |
          </div>
          <div
            class="seatClass"
          >
            economy
             class
          </div>
          <div>
            11D
          </div>
        </div>
      </div>
      <div
        class="price"
      >
        Rp 200.000
      </div>
    </div>
    <hr />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
